python/nano/src/bigdl/nano/pytorch/inference/optimizer.py

```diff
@@ -92,105 +117,202 @@ def optimize(self, model,
-
+                option = ALL_INFERENCE_ACCELERATION_METHOD[method]
+                use_ipex = option.ipex
+                use_channels_last = option.channels_last
+                accelerator = option.get_accelerator()
+                precision = option.get_precision()
                 # if precision is fp32, then we will use trace method
-                if split_method[1] == "fp32":
+                if precision == "fp32":
                     input_sample = tuple(next(iter(training_data))[:-1])
                     try:
                         if accelerator is None and use_ipex is False:
                             accelerated_model = model
                         else:
-                            # TODO: remove the logging of tracing
-                            accelerated_model = Trainer.trace(model=model,
-                                                              accelerator=accelerator,
-                                                              use_ipex=use_ipex,
-                                                              input_sample=input_sample)
-                        # TODO: 100 trial run is now fixed, we may make it adjusted intelligently.
-                        result_map[method] = {}
-
-                        def func_test(model, input_sample):
-                            model(*input_sample)
-                        result_map[method]["latency"] =\
-                            _throughput_calculate_helper(100, func_test,
-                                                         accelerated_model, input_sample)
-                        if validation_data is not None and metric is not None:
-                            result_map[method]["accuracy"] =\
-                                _accuracy_calculate_helper(accelerated_model,
-                                                           metric, validation_data)
-                        result_map[method]["model"] = accelerated_model
+                            if accelerator in ("jit", None):
+                                accelerated_model = Trainer.trace(model=model,
+                                                                  accelerator=accelerator,
+                                                                  use_ipex=use_ipex,
+                                                                  channels_last=use_channels_last,
+                                                                  input_sample=input_sample)
+                            else:
+                                accelerated_model = Trainer.trace(model=model,
+                                                                  accelerator=accelerator,
+                                                                  input_sample=input_sample,
+                                                                  # remove output of openvino
+                                                                  logging=False)
```

Comment:
why are we split the calling signature here for "jit" and None vs Onnx and openvino

Reply:
because channels_last for openvino will cause erorr in torch.export.